google-apis-core/lib/google/apis/core/storage_upload.rb

```diff
@@ -182,6 +219,54 @@ def process_response(status, header, body)
           super(status, header, body)
         end
 
+        def check_resumable_upload_status(client)
```

Comment:
We're doing more than just checking the resumable upload here. Can you please rename the method accordingly? or make the method more modular.